app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabViewHolder.kt

```diff
@@ -27,14 +27,22 @@ class RecentTabViewHolder(
 ) : ViewHolder(view) {
 
     fun bindTab(tab: TabSessionState) {
-        recent_tab_title.text = tab.content.title
-
         if (tab.content.icon != null) {
             recent_tab_icon.setImageBitmap(tab.content.icon)
         } else {
             icons.loadIntoView(recent_tab_icon, tab.content.url)
         }
 
+        // A page may take a while to retrieve a title, so let's show the url until we get one.
+        val text = if (tab.content.title.isNotEmpty()) {
+            tab.content.title
+        } else {
+            tab.content.url
+        }
```

Comment:
Maybe we should just inline this 

```suggestion
        recent_tab_title.text = if (tab.content.title.isNotEmpty()) {
            tab.content.title
        } else {
            tab.content.url
        }
```